weaviate/users/my_user_getter.go

```diff
@@ -0,0 +1,28 @@
+package users
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/weaviate/weaviate-go-client/v4/weaviate/connection"
+	"github.com/weaviate/weaviate-go-client/v4/weaviate/except"
+	"github.com/weaviate/weaviate/entities/models"
+)
+
+type MyUserGetter struct {
+	connection *connection.Connection
+}
+
+func (mug *MyUserGetter) Do(ctx context.Context) (*models.UserInfo, error) {
+	path := "/users/own-info"
+	res, err := mug.connection.RunREST(ctx, path, http.MethodGet, nil)
+	if err != nil {
+		return nil, except.NewDerivedWeaviateClientError(err)
+	}
+	if res.StatusCode == http.StatusOK {
+		var user *models.UserInfo
+		decodeErr := res.DecodeBodyIntoTarget(&user)
+		return user, decodeErr
```

Comment:
question: why do we need to pass a pointer to a nil variable? can we instead pass just `nil`?

Reply:
We can either have `var user *models.UserInfo` or `var user models.UserInfo`, but we'll need to pass `&user` to `json.Unmarshal` either way.

Here's a [Go Playground example](https://go.dev/play/p/Fxc0XUdQL3k) and the [SO post](https://stackoverflow.com/questions/35604356/json-unmarshal-accepts-a-pointer-to-a-pointer) I learned about this from

Reply:
In general, do you think clients should returns structs or pointers to structs?
I know returning pointers may save some memory, but in this case it doesn't seem like we're going to win much.

Maybe we should return `models.UserInfo` here instead of `*models.UserInfo`.

Reply:
got it, didn't know that we inside an Unmarshaler